### ISSUE TITLE
Date format / timezone issue after remote build

### ISSUE THREAD

**Issue by willduncanphoto** [gohugoio/hugo#9996]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
0.100.2
</pre>

### Does this issue reproduce with the latest release?
Yes.

When attempt to use the timezone in a `.Date.Format` for a post/page, it works locally, but when deployed, it displays the timezone as `-0700` no matter what.  And when I use just `{{ .Date }}` it displays `-0700 -0700`.

`{{ .Date.Format "January 2, 2006 3:04 PM MST" }}` and `{{ .Date }}` return the following on both Cloudflare and Netlify.

<img width="544" alt="Screenshot_20220612T005802@2x" src="https://user-images.githubusercontent.com/2657/173223289-a73accf8-a7f0-4e2c-b5f5-0d1885e651d9.png">

Locally it spits out `June 11, 2022 at 2:06 AM PDT // 2022-06-11 02:06:56 -0700 PDT` as it should.

Are there logs I can track to figure out the cause? As far as I can tell, it's a bug of some kind, but that both netlify and pages suffer from? is it a timezone/config issue on my end and I'm crazy?

**Comment by bep** [comment]
So,

This code:

```
{{ .Date.Format "January 2, 2006 3:04 PM MST" }}
```

Uses the local timezone on whatever server it's running on for any zone settings (offset, name). This is unfortunate, but not something Hugo can do anything about, which is why we added a format func:

```
{{ .Date | time.Format "January 2, 2006 3:04 PM MST" }}
```

The above will, if you have set a `timeZone` in your Hugo config (can be set per language), give more consistent results.

I suggest we take further trouble shooting at:

https://discourse.gohugo.io/

**Comment by willduncanphoto** [comment]
Thanks. I applied as you suggested, same results. That is a much better solution though, so thank you!

The timezone / fomatting are correct on the posts after the build on Cloudflare. I assumed that was set by the timezone set in the post front matter so I didn't need timezone in the config. Since I'm just trying to display the PST / PDT with the post time. I'm still not the most savvy Hugo user.

Local: 

<img width="326" alt="Screenshot_20220612T031435@2x" src="https://user-images.githubusercontent.com/2657/173228481-8491b57d-d4bb-4a81-8226-6ccb2927b9d9.png">

On Cloudflare/Netlify builds:

<img width="281" alt="Screenshot_20220612T031351@2x" src="https://user-images.githubusercontent.com/2657/173228454-17b5f23d-00a6-4e69-b471-a76eebd186e8.png">

Cloudflare was able to replicate it though. If it's not a bug, it might be a misconception in the configuration of Hugo on these services? But so far I've been unable to figure out why it's happening.

**Comment by satotake** [comment]
This might be a workaround.
Could you set `TZ` environment variable before building such as `TZ="/usr/share/zoneinfo/America/Denver"`?
(I have not tested with netlify.)

**Comment by satotake** [comment]
Tested on netlify.
See this.

- [source](https://github.com/satotake/hugo-test/tree/issue-9996)
- [preview](https://62a5dafa93d11100095111b4--stellar-sprinkles-06abfb.netlify.app/)

**Comment by techmagus** [comment]
Just adding this thread from Hugo discourse:
https://discourse.gohugo.io/t/possible-bug-mst-showing-as-0900-instead-of-jst-or-kst/37883

I currently don't have extra time to get back to it but if someone wants to do testing for GH Pages, the repo sample is:
https://github.com/techmagus/hugo-test-site

**Comment by willduncanphoto** [comment]
Copy that.  Was able to replicate it on both without the TZ set in the environmental variable. The server doesn't seem to know how to return the PDT/PST without it.

And without it, it spits back the -0700 in place of PDT.

Updated netlify.toml and set the environment variable in cloudflare and it appears to work now.

I still thing it's buggy behavior that I don't understand at all.  My timezone without the TZ configuration was correct. It's also not something that is mentioned in any tutorial or documentation / build instructions for Cloudflare documentation that I can find.

At least it can easily be mitigated.

**Comment by techmagus** [comment]
I just tested the workaround and it only works for single language/timezone sites. If a site is multilingual and the relative timezone of the language is different from the TZ environment variable, it won't work for those.

In my test site (https://hugo-test-site.pages.dev) I set the TZ env to `/usr/share/zoneinfo/Asia/Manila` and the language that matched `Asia/Manila` started to show the correct MST value which is `PST`.

However, in the JA and KO languages, it still shows `+0900` instead of `JST` and `KST` respectively, even though the system is capable of displaying the correct MST value.

Tested in Cloudflare Pages,  haven't tested in Github Pages but probably will be the same result.

---

Note: The checkmark or x-mark shown in the test site are automatically generated on build if it matches the expected field. The source for the test site is at: [GitHub - techmagus/hugo-test-site: YourOnly.One's Hugo Test Site](https://github.com/techmagus/hugo-test-site)

---

Also posted here: [Hugo Discourse thread](https://discourse.gohugo.io/t/bug-mst-showing-as-0900-instead-of-the-mst-value/37883/17?u=techmagus)
